lib/ansiblelint/__main__.py

```diff
@@ -153,7 +153,7 @@ def main() -> int:
         for match in matches:
             print(formatter.format(match))
 
-    if matches:
+    if matches and not (options.quiet or options.parseable or options.parseable_severity):
```

Comment:
I agree with `-q` but not the others.
Also, @ssbarnea is correct about the need to redirect the hint output to stderr.

There's other problems with this too:
* Adding the flag check to this if-clause changes the return code behavior effectively breaking those flags and making them trick ansible-lint into returning the successful outcome regardless of the violations present
* This change must include regression tests, both for the hint absence with `-q` and checking that adding it does not affect the return code

Reply:
I personally use parseable and parsable_severity as inputs into a larger testing framework. I'm sure others do as well. The adding of lines that do not follow the established pattern requires additional parsing that is unnecessary. STDERR is generally reserved for error messages of which this is not. It is an informational message that although provides value, is not necessary.

As for your comment on the return codes - I agree. I would propose reverting to the original if statement and creating a new if statement for this function gated by the flags.

Reply:
Alter the code and use a nested conditional approach similar to the warning feature PR currenly open:

```
if matches:
   if not options.quiet and not options.parseable:
       ...
```

That is much easier to read and works file. There is no need to look at parseable_severity -- we do not care about its value. The only case where we want to skip printing is when we are in parsable mode or quiet.

I know I proposed stderr, but that is tricky as it needs lots of testing, especially around ANSI support on stderr, which can vary.

Stderr is for outof the band messaging, logging most of the time, not only error, also debug, and "info". This message fits well in info category. Still, I cannot say for sure that either stdour or stderr are the ultimate correct stream to use for this message. Is a bit of grey area. So, when in doubt I am inclined to avoid making a change on this.

Reply:
@jbackman stderr does not exist solely for error messages. It should be used for any (often human-readable) diagnostic output so that stdout is clean and can be piped into other programs for parsing.